capstone-app/src/app/modules/history/wiki-media/wiki-media.component.ts

```diff
@@ -50,17 +50,34 @@ export class WikiMediaComponent implements OnInit, OnDestroy {
       let endIndex = firstPartOfString.indexOf("<h2>", 0);
       let startIndex = firstPartOfString.indexOf('</h2>', 0);
       let middleOfString = firstPartOfString.substring(startIndex, endIndex);
-      let finalString = middleOfString.split('href="/wiki').join('target="_blank" href="https://en.wikipedia.org/wiki');
-      finalString.split('title="Edit section: History">edit').join('>');
-      return finalString;
+      let paragraphs = middleOfString.split('<p>');
+      if (paragraphs.length > 6){
+        middleOfString = paragraphs.slice(0,5).join('');
+      }
+      this.findHrefs(middleOfString);
+      let history = middleOfString.split(/<.*?>/g).join("");
+      return history.split(/&.*?;/g).join("");
     }
     console.log("The city page was found, but unfortunately there was no history paragraph found!");
-    return text;
+    let history = text.split(/<.*?>/g).join("");
+    return history.split(/&.*?;/g).join("");
   }
 
-  setHtml(text: string) {
-    let el = document.getElementById('body');
+  findHrefs(text: string) {
+    let el = document.createElement("p");
```

Comment:
Didn't Matt discourage directly updating the DOM? Why not create an array of paragraphs and insert them into the html by including a structural directive in the .html file? This would be even easier if you created the wiki interface I mentioned that included a paragraphs key.

Reply:
I am not manipulating the dom as the elements are never added to the page. I am only using that to find all the href elements  in the response before I remove all the html tags, and turn them into a list which I set to an array in my code, which is then looped over in the html.